### PR TITLE
Added android module

### PR DESCRIPTION
**Pull request by brentwatson**
- Added `android` sub-module that extends `core` module.  
- Added `com.percolate.sdk.android.dto` package that extends parent DTOs and implements Parcelable.
- Script to generate DTOs is here: https://gist.github.com/brentwatson/e7313282dfc1368cc3e9
- Generated parcelable code using the "Android Parcelable Generator" plugin for IntelliJ:  https://github.com/mcharmas/android-parcelable-intellij-plugin
- ...this plugin did not find fields in subclasses, so I had to fork it and add an option to allow subclasses.  Opened PR to push that change upstream: https://github.com/mcharmas/android-parcelable-intellij-plugin/pull/48

Next PR will be to move our DTO unittests into this project.

Using this build in gradle:
```
compile 'com.github.percolate.percolate-java-sdk:core:462d4e63ed13ab70dc33a4dbe49265403f278b99`
compile 'com.github.percolate.percolate-java-sdk:android:462d4e63ed13ab70dc33a4dbe49265403f278b99
```

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    compileSdkVersion rootProject.ext.compileSdkVersion
+    buildToolsVersion rootProject.ext.buildToolsVersion
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testCompile group: 'junit', name: 'junit', version: '4.11'
```
Comment (msya) [inline android/build.gradle]: 4.12?

File: android/local.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# This file is automatically generated by Android Tools.
+# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
+#
+# This file must *NOT* be checked into Version Control Systems,
+# as it contains information specific to your local configuration.
+
+# location of the SDK. This is only used by Ant
+# For customization when using a Version Control System, please read the
+# header note.
+sdk.dir=/home/brent/Android/Sdk
```
Comment (msya) [inline android/local.properties]: Should this file be in .gitignore?

File: android/build.gradle (hunk 1 above)
Comment (brentwatson) [inline android/build.gradle]: Done in branch for next PR where I add unit tests: https://github.com/percolate/percolate-java-sdk/commit/15c0b4b31dc53506557ebf9e4632b4f9eaddb34f

File: android/local.properties (hunk 2 above)
Comment (brentwatson) [inline android/local.properties]: Yup.  `unit_test_setup` branch has 2 commits:
- Removing local.properties
- Ignoring local.properties

**Pull request merged**